Adn/reisolate itinerary

alexandredamiron opened the pull request:
This refacto prepare efforts to completely isolate the whole itinerary component and show it into storybook

Khoyo committed:
core: add signaling system block checking

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

bloussou committed:
Add `CONTRIBUTING.md`

transifex-integration[bot] committed:
Apply translations in en

100% translated for the source file 'front/public/locales/fr/map-key.json'
on the 'en' language.

Khoyo committed:
Revert "front: block state: display on simulation's map"

This reverts commit cf2ecbab6ed8ac8b327eab0c5fbc957df585ec64.

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

leovalais committed:
editoast:search: add support of arrays in the query language

Khoyo committed:
core: add mock tvm signaling

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

Khoyo committed:
core: fix blocks with no zonepath

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

leovalais committed:
editoast:search: remove error variant InContext

sim51 committed:
front: using types generated by rtk

sim51 committed:
front: editor - load editor schema by using rtk query

sim51 committed:
front: migrate saveEditor & getSwitchTypes by using RTK

sim51 committed:
front: remove unused

sim51 committed:
front: editor - refacto editoastToEditorEntity and railjson type

leovalais committed:
api:search: add signal search table migration

clarani committed:
front: GET - fix coordinated cursors

younesschrifi committed:
editoast: migrate scenario model

Khoyo committed:
core: load kotlin infra when we load infras

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

leovalais committed:
editoast:search: simplify compound types parsing in search.yml

leovalais committed:
editoast:search: fix string binding order

younesschrifi committed:
editoast: add post scenario endpoint

younesschrifi committed:
editoast : add delete endpoint for scenario

anisometropie committed:
front: fix warnings & ts

front: ts & warnings

errors numbers 140

fixes

f

import

warning

fix import

clarani committed:
editoast: move light_rolling_stock to /models
- move LightRollingStockModel to /models
- create LightRollingStock schema
- add tests

clarani committed:
editoast: move RS livery to ./models/rolling_stock
- move file
- implement Model trait for rollingStockLivery
- clean tests

nicolaswurtz committed:
front: bugfix on chipsncf & trainlabel (use new selectors)

nicolaswurtz committed:
front: i18n: use infrastructures count for plurals

alexandredamiron committed:
front: change default allowance settings in stdcm

# Conflicts:
#	front/src/applications/stdcm/views/OSRDStdcmResults.jsx

# Conflicts:
#	front/src/applications/stdcm/views/OSRDStdcmResults.jsx
#	front/src/applications/stdcm/views/StdcmRequestModal.jsx

alexandredamiron committed:
front: no unused var

younesschrifi committed:
editoast: add get endpoint for scenario

anisometropie committed:
front: refacto offset function

rename

clarani committed:
api: update rolling_stock_image and rolling_stock_livery models to use document

leovalais committed:
editoast:search: add signal search criteria logical_types

leovalais committed:
editoast:search: add geographic info to signal search response

multun committed:
api: remove route_{begin,end}_occupancy

This API is a legacy from the time where routes, zones and blocks were
unified into a single weird blob. It's gone now.

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

nicolaswurtz committed:
front: split some locales files

nicolaswurtz committed:
front: i18n: add menu entry & modal to change language

Khoyo committed:
core: make signal drivers more resilient

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

Khoyo committed:
core: split the signal simulation out of the standalone sim endpoint

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

younesschrifi committed:
editoast : add update endpoint for scenario

multun committed:
core: migrate standalone sim to the new signaling simulator

Co-Authored-By: Victor "multun" Collod <victor.collod@epita.fr>
Co-Authored-By: Younes "Khoyo" Khoudli <younes.khoudli@epita.fr>

clarani committed:
editoast: update rolling_stock_livery model to replace RS images by document

flomonster committed:
schemas: replace applicable directions to direction for buffer stops

Yohh committed:
front: stdcm: set results under the map; set display of inputs and results in 1/3-2/3; adapt map height before simulation; make map and GEV deployable after simulation; edit simulation.json and manageTrainSchedule.json

- convert OSRDStdcmResults.jsx to tsx

- remove simulation consolidation from OSRDStdcmResults.tsx; clean display in case no train id is selected; add TODO comment to refacto request status with RTK

- remove GET and GET cards before simulation; edit labels for standardAllowancesDefault

- clean nested div & empty classes in standardAllowancesDefault.jsx

- adapt map-size after simulation

- fix map origin/destination, fix GEV height

leovalais committed:
editoast:search: simplification of type specs in search.yml

eckter committed:
stdcm: edges don't go further than the stop

leovalais committed:
editoast:search: remove arg bind_pos in sqlquery

alexandredamiron committed:
front: get advance on Origin and Vias

flomonster committed:
core: adapt generator for railjson 3.2.0

transifex-integration[bot] committed:
Apply translations in en

100% translated for the source file 'front/public/locales/fr/operationalStudies/study.json'
on the 'en' language.

transifex-integration[bot] committed:
Apply translations in en

100% translated for the source file 'front/public/locales/fr/map-search.json'
on the 'en' language.

transifex-integration[bot] committed:
Apply translations in en

100% translated for the source file 'front/public/locales/fr/rollingstock.json'
on the 'en' language.

transifex-integration[bot] committed:
Apply translations in en

100% translated for the source file 'front/public/locales/fr/home/home.json'
on the 'en' language.

dependabot[bot] committed:
editoast: bump openssl from 0.10.47 to 0.10.48 in /editoast

Bumps [openssl](https://github.com/sfackler/rust-openssl) from 0.10.47 to 0.10.48.
- [Release notes](https://github.com/sfackler/rust-openssl/releases)
- [Commits](https://github.com/sfackler/rust-openssl/compare/openssl-v0.10.47...openssl-v0.10.48)

---
updated-dependencies:
- dependency-name: openssl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

flomonster committed:
editoast: add cargo chef docker image

clarani committed:
front: clean GEV

anisometropie committed:
front: fix coordinates crash

anisometropie committed:
front: fix graph hovering crash

Akctarus committed:
front: power restrictions: add selector

clarani committed:
front: fix-bug: display rolling stock images

Akctarus committed:
front: power restrictions: update reducers

younesschrifi committed:
editoast: front: uses editoast endpoints instead of api endpoints for study and scenario

Akctarus committed:
front: power restrictions: handle no data

alexandredamiron committed:
front: rsiolate fro DisplayItineary

bloussou committed:
core: Remove `python_interactive_client/`

alexandredamiron committed:
front: isolate ModalSuggeredVias

younesschrifi committed:
editoast : add suffix _id

leovalais committed:
editoast: add osrd_infra_pathmodel and GET /pathfinding/{id}

leovalais committed:
editoast: add DELETE /pathfinding/{id}

eckter committed:
core: fix envelope discontinuity when computing accelerations

younesschrifi committed:
editoast: adapt infra to new model (retrieve and delete)

SharglutDev committed:
front: releaseinformations: remove releaseInformations.scss; change .informations to .informations-modal; create _relaseInformations.scss in /styles

- replace modalheadersncf children by withclosebutton props

- navbarsncf  add lg size to props when calling openmodal; releaseinformation change display to have 2 columns on large screens; licenseattribution add scroll only for this column

alexandredamiron committed:
front: refacto to isolate displayVias index

Akctarus committed:
front: power restrictions: finishing selector

front: powerRestriction: some light refacto kiss kiss

front: power restrictions: add "no selection" option

front: power restrictions: updating names

bloussou committed:
tests: Remove `setup_dummy_db` (#3690)

flomonster committed:
editoast: fix missing buffer stop warnings

bloussou committed:
editoast: `ntest` to `rtest`

flomonster committed:
editoast: add buffer stop odd location warning

younesschrifi committed:
editoast : adapt infra model (create)

clarani committed:
editoast: migrate POST /rolling_stock and implement model for rolling_stock (#3639)

closes #3499

flomonster committed:
editoast: add list trait for model

younesschrifi committed:
editoast: add update and list model for infra

clarani committed:
front: add suffix _id to scenario's timetable, infra, electrical_profil_set

younesschrifi committed:
editoast: api: fix study date type and fix electrical_profile patch field

flomonster committed:
editoast: fix duplicate rolling stock schema

Khoyo committed:
cord: update openapi.yml

flomonster committed:
editoast: bump dependencies

- **tempfile** from 3.4.0 to 3.5.0 (close #3728)
- **serde** from 1.0.158 to 1.0.159 (close #3715)
- **async-trait** from 0.1.67 to 0.1.68 (close #3703)
- **serde_json** from 1.0.94 to 1.0.95 (close #3702)
- **image** from 0.24.5 to 0.24.6 (close #3701)

nicolaswurtz committed:
front: fix notification above all & map add train too big

Akctarus committed:
front: power restrictions: connecting front to api

Akctarus committed:
python: update middleWare api

Yohh committed:
front: STDCM: set RollingstockSelector in one line (#3742)

close #3610

---------

Co-authored-by: alexandredamiron <alexandredamiron@gmail.com>

alexandredamiron committed:
front; correct tsx on modalSuggerdVias

alexandredamiron committed:
front: removeUseless map symbol

flomonster committed:
editoast: fix openapi for the clone infra endpoint

Yohh committed:
front: STDCM: fix map height without effect in operational studies

clarani committed:
front: remve warnings from DisplayMissingInfo

bloussou committed:
tests: Make e2e tests call the backend (#3672)

nicolaswurtz committed:
front: releaseinformations: light refacto, add links

clarani committed:
editoast: front: adapt payload to create and update scenario

alexandredamiron committed:
front: use new HOC on sppedTimeChart on stdcm

bloussou committed:
front: tests: Add a `yarn e2e-tests` and display playwright outputs (#3770)

bloussou committed:
core: tests: Isolate Railjson generator (#3624)

clarani committed:
Fix integration tests

younesschrifi committed:
editoast: front: sort projects, studies and scenarios

anisometropie commented on front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/Destination.tsx:
```diff
@@ -21,20 +22,58 @@ import { makeEnumBooleans } from 'utils/constants';
+}
+
+export function withOSRDData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
```
better to have named function for the react dev tool tree

alexandredamiron replied:
It has been renamed

anisometropie commented on front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/Destination.tsx:
```diff
@@ -21,20 +22,58 @@ import { makeEnumBooleans } from 'utils/constants';
+    const destination = useSelector(getDestination);
+    const destinationDate = useSelector(getDestinationDate);
+    const destinationTime = useSelector(getDestinationTime);
+    const dispatch = useDispatch();
```
Why do move dispatch out the the `Destination` component ?

alexandredamiron replied:
because the component is supposed to be able to work with no store (storybook)

anisometropie commented on front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/Origin.tsx:
```diff
@@ -29,24 +29,75 @@ import {
+  dispatch?: Dispatch;
+}
+
+export function withOSRDData<T>(Component: ComponentType<T>) {
```
Every HOC in the app is named `withOSRDData`. We need to know which one we’re using, `withOSRDSimulationData` or something like that

anisometropie commented on front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/Vias.tsx:
```diff
@@ -8,17 +8,28 @@ import { useTranslation } from 'react-i18next';
-function Vias(props: ViasProps) {
-  const { zoomToFeaturePoint, viaModalContent } = props;
-  const vias = useSelector(getVias);
+export function withOSRDData<T>(Component: ComponentType<T>) {
```
same thing, `withGeoJSONData` ? or `withPathData` ?

anisometropie commented on front/src/applications/operationalStudies/components/ManageTrainSchedule/Itinerary/DisplayItinerary/index.tsx:
```diff
@@ -14,19 +17,45 @@ interface DisplayItineraryProps {
 
-export default function DisplayItinerary(props: DisplayItineraryProps) {
-  const { zoomToFeaturePoint, zoomToFeature, viaModalContent } = props;
+export function withOSRDData<T>(Component: ComponentType<T>) {
```
same

Commit:
front: update osrd_conf_schedule power_restriction ranges

bloussou committed:
ci: use `setup-python` poetry cache (#3783)

nicolaswurtz committed:
front: typescripting: ChipsSNCF

sim51 committed:
[front] fix #1567

sim51 committed:
[front] Body at 100vh to fix #3753

sim51 committed:
[front] Fix #3751 - Adding style for DropdownSNCF

clarani committed:
front: fix bug 3796 (#3798)

closes #3796 

This PR replaces scenario.infra by scenario.infra_id in Scenario.jsx

Khoyo committed:
core: fix zone occupancy computation on train with stops

bloussou committed:
front: tests: make e2e tests work when running on a dirty env (#3782)

sim51 committed:
[front] refacto modal #3493

Khoyo committed:
core: make the zone after a train path be clear

nicolaswurtz committed:
front: remove useless svg

sim51 committed:
[front] #1567 - avoid blink effect on tooltip

flomonster committed:
editoast: disable idle db connections

flomonster committed:
editoast: speedup retrieve railjson endpoint

- Split db queries

nicolaswurtz committed:
front: repair detectors layer

multun committed:
core: lower infra download memory usage

sim51 committed:
[front] Remove commented code

sim51 committed:
[front] dropdown and arrow #3750

nicolaswurtz committed:
front: no-shadow: error

flomonster committed:
editoast: fix memory consumption retrieving railjson

flomonster committed:
core: bump gradle and fix to java 17 (lts)

- Bump gradle from 7.5 to 8.0
- Fix jre to 17
- Use eclipse temurin instead of amazoncorretto

clarani committed:
front: use document endpoint to fetch rolling_stock images
- create a reusable function to fetch document
- use editoast endpoints to fetch rolling_stocks
- use document endpoint to fetch rolling_stock images

leovalais committed:
editoast: add TrackSectionModel & geos usage for geometry operations

alexandredamiron committed:
front: naming Hoc func

jacomyal committed:
editor: fine tunes overlay related to #1567

jacomyal committed:
editor: restores X button on disabled ConfirmModal

Since the modal background click is never disabled, it seems weird to
have the cross button disappearing.

flomonster committed:
editoast: move tempfile to dev-dependency

`tempfile` crate is only used for testing

flomonster committed:
editoast: adapt get railjson endpoint

Remove the `exclude_extensions` query param.

flomonster committed:
core: adapt request to retrieve a railjson

flomonster committed:
railjson_generator: disable route creation between two buffer stops

anisometropie reviewed (dismissed):
tested and validated

flomonster committed:
editoast: bump dependencies

- **serde_yaml** from 0.9.19 to 0.9.21 (close #3841)
- **redis** from 0.22.3 to 0.23.0 (close #3830)
- **spin** from 0.9.7 to 0.9.8 (close #3785)
- **json-patch** from 0.3.0 to 1.0.0 (close #3776)
- **futures** from 0.3.27 to 0.3.28 (close #3755)

alexandredamiron committed:
Adn/condensed speed limit (#3769)

A typescripted version of SpeedLimitSelectors (gabarits for speedLimit).
Also Isolation, and a story for storyBook

![Capture d’écran 2023-03-31 à 12 12
52](https://user-images.githubusercontent.com/765626/229092718-f07fda62-947a-4f85-8f3d-fe9812790895.png)

Uriel-Sautron committed:
front: map search: Search into signals
